src/ether_nethub.c

```diff
@@ -0,0 +1,599 @@
+ *	ether_reset(args) 175/73/0
+ *	reset ether controller and disable receipt of packets
+ **********************************************************************/
+LispPTR ether_reset(LispPTR args[])
+{
+  if (nethubHost == NULL) {
+    return (NIL);
+  }
+  log_debug(("ether_reset() - begin\n"));
+  
+  if (ether_fd < 0) {
+    log_debug(("ether_reset() - end -> NIL\n\n"));
+    log_info(("ether_reset() -> NIL\n"));
+    return (NIL);
+  }
+  ether_bsize = 0; /* deactivate receiver */
+  log_debug(("ether_reset() - end -> ATOM_T\n\n"));
+  log_info(("ether_reset() -> ATOM_T\n"));
+  return (ATOM_T);
+}
+
+
+/************************************************************************/
+/*									*/
+/*			  e t h e r _ g e t (175/74/2)			*/
+/*									*/
+/*	Set up the Ethernet driver to receive a packet.  The driver	*/
+/*	first tries to read any pending packet from the net, and if	*/
+/*	there is one, ether_get returns T.  If there is no pending	*/
+/*	packet, the failing read sets us up to get an interrupt when	*/
+/*	a packet DOES arrive, and ether_get returns NIL.		*/
+/*									*/
+/*		args[0] Length of the buffer we're passed		*/
+/*		args[1] LISP address of a packet buffer			*/
+/*									*/
+/*	sets ether_buf to the buffer address, for check_ether's use	*/
+/*	sets ether_bsize to the buffer size.  ether_bsize>0 means	*/
+/*	it's OK to read packets from the network on interrupt.		*/
+/*									*/
+/************************************************************************/
+LispPTR ether_get(LispPTR args[])
+{
+  if (nethubHost == NULL) {
+    return (NIL);
+  }
+  log_debug(("ether_get() - begin\n"));
+
+  u_char *target = (u_char *)NativeAligned2FromLAddr(args[1]);
+  int maxByteCount = 2 * (0xFFFF & args[0]); /* words to bytes */
+  log_debug(("  target = 0x%016lX maxBytecount: %d bytes\n", (unsigned long)target, maxByteCount));
+
+  sigset_t signals;
+  sigemptyset(&signals);
+  sigaddset(&signals, SIGIO);
+  sigprocmask(SIG_BLOCK, &signals, NULL);
+
+  ether_buf = target;
+  ether_bsize = maxByteCount;
+
+  int receivedBytes = recvPacket();
+
+  sigprocmask(SIG_UNBLOCK, &signals, NULL);
+
+  log_debug(("ether_get() ::  receivedBytes: %d bytes\n", receivedBytes));
+  if (receivedBytes <= 0) {
+    log_debug(("ether_get() - end -> NIL\n\n"));
+    return (NIL);
+  }
+
+  log_debug(("ether_get() - end -> ATOM_T\n\n"));
+  log_info(("ether_get(): received packet, len = %d bytes\n", receivedBytes));
+  return (ATOM_T);
+}
+
+
+
+/**********************************************************************
+ *	ether_send(args) 175/75/2 max_words,buffer_addr
+ *	send a packet
+ **********************************************************************/
+LispPTR ether_send(LispPTR args[])
+{
+  if (nethubHost == NULL) {
+    return (NIL);
+  }
+  log_debug(("ether_send() - begin\n"));
+
+  u_char *source = (u_char *)NativeAligned2FromLAddr(args[1]);
+  int byteCount = 2 * (0xFFFF & args[0]); /* words to bytes */
```

Comment:
I'll be looking to fix this in the original ether code as well -- this is NOT how one is supposed to access a Lisp numeric value -- I believe there's no guarantee that it's a SMALLP, the code may need to fetch a FIXP.
`#include "lisp2c.h"` and `n = LispInt2CInt(args[0]);`

Reply:
done (it's `lisp2cdefs.h` and `LispIntToCInt`)